packages/astro/src/core/build/generate.ts

```diff
@@ -148,7 +148,10 @@ export async function generatePages(opts: StaticBuildOptions, internals: BuildIn
 			if (pageData.route.prerender) {
 				const ssrEntryURLPage = createEntryURL(filePath, outFolder);
 				const ssrEntryPage = await import(ssrEntryURLPage.toString());
-				if (opts.settings.config.build.split) {
+				if (
+					opts.settings.config.build.split ||
+					opts.settings.adapter?.adapterFeatures?.functionPerRoute
+				) {
```

Comment:
Damn, good catch! 

```suggestion
				if (
				  // TODO: remove in Astro 4.0
					opts.settings.config.build.split ||
					opts.settings.adapter?.adapterFeatures?.functionPerRoute
				) {
```

Leaving a suggestion, to add a `TODO` comment. We are doing this for all deprecated config options.